### PR TITLE
Add Postgres 16 projects, remove Postgres 10 projects, switch from Postgres 14.8 to Postgres 14

### PR DESCRIPTION
**Pull request by SwooshyCueb**
In service of irods/irods#7382
Addresses #211
Addresses #212

### CODE REVIEW

**Review by alanking — COMMENTED** [review]
Looking good!

File: irods_testing_environment/odbc_setup.py [hunk 1]
```diff
@@ -15,55 +15,90 @@ def configure_postgres_odbc_driver(csp_container, odbc_driver):
...
 
     Argument:
     csp_container -- docker container on which the iRODS catalog service provider is running
     odbc_driver -- path to local archive file containing the ODBC driver package
     """
     configure_postgres_odbc_driver(csp_container, odbc_driver)
 
-def configure_odbc_driver_debian_12_postgres_148(csp_container, odbc_driver):
+def configure_odbc_driver_debian_12_postgres_16(csp_container, odbc_driver):
     """Configure ODBC driver for postgres 14.8 on debian 12.
```
Comment (trel) [inline irods_testing_environment/odbc_setup.py:102]: postgres 14.8 doesn't match function name of postgres 16

File: irods_testing_environment/database_setup.py [hunk 2]
```diff
@@ -171,7 +171,11 @@ def grant_privileges(self, database, username):
         database -- name of the database on which privileges are being granted
         username -- name of the user for whom privileges are being granted
         """
-        return self.execute_psql_command('grant all privileges on database \\\"{0}\\\" to {1};'
+        ec = self.execute_psql_command('grant all privileges on database \\\"{0}\\\" to {1};'
+            .format(database, username))
+        if ec is not 0:
+            return e
```
Comment (korydraughn) [inline irods_testing_environment/database_setup.py:177]: `e` -> `ec`?

File: irods_testing_environment/odbc_setup.py (hunk 1 above)
Comment (SwooshyCueb) [reply irods_testing_environment/odbc_setup.py]: Fixed

File: irods_testing_environment/database_setup.py (hunk 2 above)
Comment (SwooshyCueb) [reply irods_testing_environment/database_setup.py]: Fixed

**Comment by SwooshyCueb** [comment]
Ready for review

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Let's get one more approval before pounding.

**Review by alanking — APPROVED** [review]
Looks good. # it.

**Pull request merged**